src/hardware/driver/interlock.rs

```diff
@@ -0,0 +1,97 @@
+///! Driver interlock functionality
+///!
+///! Driver features an interlock to ensure safe co-operation with other devices.
+///! The interlock is implemented via MQTT. See [Interlock] for details about the MQTT interface.
+use fugit::ExtU64;
+use heapless::String;
+use miniconf::Miniconf;
+
+#[derive(Clone, Copy, Debug, Miniconf)]
+pub struct Interlock {
+    /// Interlock topic. Publishing "true" onto this topic renews the interlock timeout.
+    /// Publishing "false" trips the interlock.
+    ///
+    /// # Path
+    /// `interlock`
+    ///
+    /// # Value
+    /// "true" or "false"
+    interlock: bool,
+
+    /// Set interlock to armed/disarmed.
+    ///
+    /// # Path
+    /// `armed`
+    ///
+    /// # Value
+    /// "true" or "false"
+    armed: bool,
+
+    /// A positive flank from "false" to "true" clears the interrupt.
+    ///
+    /// # Path
+    /// `armed`
+    ///
+    /// # Value
+    /// "true" or "false"
+    clear: bool,
+
+    /// Interlock timeout in milliseconds.
+    ///
+    /// # Path
+    /// `timeout`
+    ///
+    /// # Value
+    /// "true" or "false"
+    timeout: u64,
+}
+
+impl Default for Interlock {
+    fn default() -> Self {
+        Self {
+            interlock: false,
+            armed: true,
+            clear: false,
+            timeout: 1000,
+        }
+    }
+}
+
+pub enum Handle {
+    Spawn(fugit::Duration<u64, 1, 10000_u32>),
+    Reschedule(fugit::Duration<u64, 1, 10000_u32>),
+    Cancel,
+    Idle,
+}
+
+pub fn handle(
+    path: String<64>,
+    handle_is_some: bool,
+    new: Interlock,
+    old: Interlock,
+) -> Option<Handle> {
+    let path = path.as_str();
+    if path == "interlock/interlock"
+        || path == "interlock/armed"
+        || path == "interlock/clear"
+    {
+        let cleared = !old.clear && new.clear;
+        match (handle_is_some, new.armed, cleared, new.interlock) {
```

Comment:
My suspicion is that one or two (layers of) `if`s (with `return` if you want) would be sufficient to cover this in a more readable way.